pystac/layout.py

```diff
@@ -85,7 +95,7 @@ def __init__(
         self.defaults = defaults or {}
 
         # Generate list of template vars
-        template_vars: List[str] = []
+        template_vars = []
```

Comment:
Was this annotation a problem?

Reply:
This was actually a mistake. I was trying to remove annotations that were now redundant with the instance variables annotations, but inadvertently removed this as well.

However, this annotation may not be necessary since `mypy` has no problem inferring the type.

Reply:
Getting rid of redundant annotations is good. I approve of this message. :grin:

Reply:
@schwehr Unless this is causing problems with other type checkers I'm going to continue to leave this annotation out.